sermant-plugins/sermant-tag-transmission/tag-transmission-feign11.x-plugin/src/main/java/io/sermant/tag/transmission/rpc/interceptors/FeignClientTagTransmissionInterceptor.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2025-2025 Sermant Authors. All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.tag.transmission.rpc.interceptors;
+
+import feign.Request;
+import io.sermant.core.common.LoggerFactory;
+import io.sermant.core.plugin.agent.entity.ExecuteContext;
+import io.sermant.core.utils.CollectionUtils;
+import io.sermant.core.utils.ReflectUtils;
+import io.sermant.core.utils.tag.TrafficTag;
+import io.sermant.core.utils.tag.TrafficUtils;
+import io.sermant.tag.transmission.config.strategy.TagKeyMatcher;
+import io.sermant.tag.transmission.interceptors.AbstractClientInterceptor;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The client enhancement class, initiates the feign request method
+ *
+ * @author chengyouling
+ * @since 2025-02-25
+ */
+public class FeignClientTagTransmissionInterceptor extends AbstractClientInterceptor<Map<String, Object>> {
+    private static final Logger LOGGER = LoggerFactory.getLogger();
+
+    @Override
+    protected ExecuteContext doBefore(ExecuteContext context) {
+        Object argument = context.getArguments()[0];
+        if (argument instanceof Request) {
+            Request request = (Request) argument;
+            LOGGER.log(Level.FINE, "feign client request url: {0}", request.url());
+            Map<String, Object> headers = new HashMap<>(request.headers());
+            injectTrafficTag2Carrier(headers);
+            ReflectUtils.setFieldValue(request, "headers", Collections.unmodifiableMap(headers));
+            LOGGER.log(Level.FINE, "after refactor feign client request headers: {0}", request.headers());
+        }
+        return context;
+    }
+
+    @Override
+    protected ExecuteContext doAfter(ExecuteContext context) {
+        return context;
+    }
+
+    @Override
+    protected void injectTrafficTag2Carrier(Map<String, Object> headers) {
+        TrafficTag trafficTag = TrafficUtils.getTrafficTag();
+        if (trafficTag == null || trafficTag.getTag() == null) {
+            return;
+        }
```

Comment:
The tag information has been checked in the before method and does not need to be checked again.

Reply:
fixed